src/InfrastructureLayer/Settings/Settings.cpp

```diff
@@ -24,31 +24,35 @@ namespace
 Settings::Settings(QString filepath)
     : settings_(filepath, SETTINGS_FILE_FORMAT)
 {
+    if (settings_.value(IP_ADDRESS).isNull())
```

Comment:
This will only create a default config file when the ip address is not set. I think we should set the default values for each undefined value, and leave them alone if they are already set. Currently if I set something like `exchangeName` and start the software the value will be overwritten with the default one, which kind of defeats the purpose of the config file.